cli/command/stack/deploy_composefile.go

```diff
@@ -320,6 +322,13 @@ func deployServices(
 			if sendAuth {
 				updateOpts.EncodedRegistryAuth = encodedAuth
 			}
+
+			if image != service.Spec.TaskTemplate.ContainerSpec.Image {
```

Comment:
I'm not sure this check makes sense. It means that if the digest wasn't pinned before, we won't pin the digest (or add platform information) going forward. Something like a temporary registry error would cause pinning to be turned off for future runs as well.